lib/mayaHydra/mayaPlugin/renderOverride.cpp

```diff
@@ -802,6 +804,35 @@ MStatus MtohRenderOverride::Render(
        _displayStyleSceneIndex->SetRefineLevel({true, delegateParams.refineLevel});
     }
 
+    // Update "Show" menu filters
+    {
+        auto objectExclusions = framecontext->objectTypeExclusions();
+
+        static const TfTokenVector polygonFilters = { 
+            FvpPruningTokens->meshes, 
+            FvpPruningTokens->capsules, 
+            FvpPruningTokens->cones, 
+            FvpPruningTokens->cubes, 
+            FvpPruningTokens->cylinders, 
+            FvpPruningTokens->spheres
+        };
+        static const std::map<MUint64, TfTokenVector> mayaFiltersToFvpPruningTokens = {
+            { MHWRender::MFrameContext::kExcludeMeshes, polygonFilters },
+            { MHWRender::MFrameContext::kExcludeNurbsCurves, {FvpPruningTokens->nurbsCurves} },
+            { MHWRender::MFrameContext::kExcludeNurbsSurfaces, {FvpPruningTokens->nurbsPatches} }
+        };
+
+        for (auto [mayaFilter, fvpPruningTokens] : mayaFiltersToFvpPruningTokens) {
+            for (const auto& fvpPruningToken : fvpPruningTokens) {
+                if (objectExclusions & mayaFilter) {
+                    _pruningSceneIndex->EnableFilter(fvpPruningToken);
```

Comment:
Minor: if you'd had an interface with EnableFilter(bool, token) you could have save a couple of lines of code and a bit of awkwardness here:
`_pruningSceneIndex->EnableFilter(objectExclusions & mayaFilter, fvpPruningToken)`

Reply:
Made the Enable/Disable methods private and changed the public API to a SetFilterStatus method. Also removed the bool return value since that wasn't really useful.